src/components/Slider2.js

```diff
@@ -11,12 +12,23 @@ const Slider_2 = () => {
     slidesPerView: 1.5,
   };
 
+  const { productCountReSet, isInCard } = useAddToCard();
+
+  const productCountReSetHandler = (id) => () => {
+    if (!isInCard(id)) {
+      productCountReSet(0);
+    }
+  };
   return (
     <section className="products-container-list-2">
       <Swiper {...params}>
         {data.map(({ img1, title, price, id }) => {
           return (
-            <Link to={`/item/${id}`} key={id}>
+            <Link
+              to={`/item/${id}`}
+              key={id}
+              onClick={productCountReSetHandler(id)}
```

Comment:
why we should reset this? I am clicking to see product detail, there is no need to reset the product count in my card.